Feature coke

yonsaber opened the pull request:
Added coal coke along with a coal coke block. 

Coal coke is only available from the creative menus but can be crafted into a block. Both are fuels but the burn time needs to be changed ideally. 

For now the burn times are 5 minutes and 10 minutes respectively.

Textures need to be improved for sure but they work for now.

What do you think @Hanse00?

Commit:
Created coal coke item

Created base item with texture

Commit:
Updates

Removed un-needed methods. Made in to a fuel

Commit:
Coal Coke block

Added a coal coke block, can also be used as fuel

Hanse00 commented on src/main/java/dk/philiphansen/craftech/items/ModItems.java:
```diff
@@ -0,0 +1,45 @@
+	public static Item coalCoke;
+	
+	public static void init() {
+		coalCoke = new ItemCoalCoke().setUnlocalizedName("coalCoke");
```
Remove `.setUnlocalizedNamed` and call `setName` in the constructor of the item. Also use a variable from ItemInfo rather than hard coded string.

yonsaber replied:
I forgot to add in the variable I had in there already.

Hanse00 commented on src/main/java/dk/philiphansen/craftech/items/ModItems.java:
```diff
@@ -0,0 +1,45 @@
+	
+	public static void initCrafting() {
+		//Coal Coke
+		GameRegistry.addRecipe(new ItemStack(ModBlocks.blockCoalCoke, 1), "CCC", "CCC", "CCC", 'C', coalCoke);
```
I think it would be more clear if you used the same format here as I did for the block crafting

```
        GameRegistry.addShapedRecipe(new ItemStack(blockLimestoneBrick, 4), new Object[] {
            "XX",
            "XX",
            'X', blockLimestone
        });
```

That way you can easily picture the grid and see the setup. Doesn't matter so much for this as it's clear that it's three rows of the same item, but overall it's more clear.

yonsaber replied:
Makes sense, easier reading makes everything easier in the long run!

Hanse00 commented on src/main/java/dk/philiphansen/craftech/reference/BlockInfo.java:
```diff
@@ -28,4 +28,7 @@
 	public static final String LIMESTONE_BRICK_TEXTURE = "limestonebrick";
+	
+	public static final String COALCOKE_BLOCK_NAME = "blockCoalCoke";
+	public static final String COALCOKE_BLOCK_TEXTURE = "coalCokeBlock";
```
I'd use a lower case name, it's the standard for minecraft textures and I think we should follow it.

What do you think about `coke_block.png` instead? Reflects Minecraft's `coal_block.png` well I think.

yonsaber replied:
I'm more than happy to change it over

Hanse00 commented on src/main/java/dk/philiphansen/craftech/reference/ItemInfo.java:
```diff
@@ -0,0 +1,25 @@
+	public static final String TEXTURE_LOCATION = "craftech";
+	
+	public static final String COALCOKE_NAME = "coalCoke";
+	public static final String COALCOKE_TEXTURE = "coalCoke";
```
Same thing for the texture here, I'd just go with `coke.png`

yonsaber replied:
Can do.

Hanse00 commented on src/main/java/dk/philiphansen/craftech/items/ItemCoalCoke.java:
```diff
@@ -0,0 +1,35 @@
+public class ItemCoalCoke extends Item {
+	public ItemCoalCoke() {
+		super();
+		this.setCreativeTab(CreativeTabs.tabMisc);
```
Coal and Charcoal are in the Materials tab, move it in there maybe?

yonsaber replied:
Okay, will make that change

Hanse00 commented:
Other than the line comments I've made, I suggest moving both the fuel handlers into one, no need to have two for it.

And I imagine you could replace `(fuel.getUnlocalizedName().equals(ModBlocks.blockCoalCoke.getUnlocalizedName().toString()))`
with something like `(fuel.getItem == ModBlocks.blockCoalCoke)` or maybe `(fuel.getBlock == Modblocks.blockCoalCoke)`

Once those things are done, I think we can merge it in.

yonsaber commented:
With the `(fuel.getUnlocalizedName().equals(ModBlocks.blockCoalCoke.getUnlocalizedName().toString()))` issues, I will check if `fuel.getBlock` is a thing, if not then we might be stuck with that long string because `(fuel.getItem == ModBlocks.blockCoalCoke)` will throw an error

Hanse00 commented:
Try `(fuel.getItem == Item.getItemFromBlock(ModBlocks.blockCoalCoke))` then.

Commit:
Renamed / Moved textures

Commit:
Moved coke item naming

Commit:
Changed crafting format
The crafting for coke block should now be much more readable.

Commit:
Moved coke to materials tab.

Commit:
Moved to unified fuel handler
There's now one fuel handler for all our custom fuels, every tweak to fuels will go in there.

Hanse00 commented:
Alright, here are the fixes. I think it's all good now. What about you?

yonsaber commented:
I am happy with all of that, if I spot any issues I can quickly fix them :)

Hanse00 commented:
We've got coke and limestone now, this is going places :smile:
